pkg/webhooks/regularuser/regularuser.go

```diff
@@ -225,6 +231,18 @@ func isCustomDomainAuthorized(request admissionctl.Request) bool {
 		request.Kind.Group == customDomainGroup)
 }
 
+// isNetNamespaceAuthorized check if request is authorized for NetNamespace CR
+func isNetNamespaceAuthorized(request admissionctl.Request) bool {
+	fmt.Printf("%s\n", request.Name)
+	return ((utils.SliceContains("cluster-admins", request.UserInfo.Groups) ||
+		utils.SliceContains("dedicated-admins", request.UserInfo.Groups)) &&
+		request.Kind.Kind == netNamespaceKind &&
+		request.Kind.Group == netNamespaceGroup &&
+		request.Name != "" &&
+		!namespace.PrivilegedNamespaceRe.Match([]byte(request.Name)) &&
+		!namespace.BadNamespaceRe.Match([]byte(request.Name)))
```

Comment:
Can we tidy this, please?

Reply:
Sure, I've added a helper function `isNetNamespaceValid()`

Reply:
What I'm looking for is to break these so they're not compound ifs with ors, ands, and equality checks in. It's hard to read and parse.

Consider, for example an improvement:

```go
if request.Name == "" {
	return false
}
// some other check
```

Even better would be to encapsulate the semantic meaning behind what it means for `request.Name` to be equal to an empty string. As a human, I can't understand _why_ it's needed, but if it was encapsulated in a function, with comments around it, it will help future contributors understand.

We can trust the compiler to optimize the logic statements so the computers can run them, let's focus on optimizing for humans to read them.

Reply:
The ask is a little out of scope and not consistent with the other functions, but I went ahead and changed `isNetNamespaceValid` to be a little more readable.